app/experimenter/docs/adrs/0002-use-react-bootstrap-cra-gql-for-nimbus.md

```diff
@@ -0,0 +1,113 @@
+# Use React, Bootstrap, CRA, and GQL for the Nimbus Front-End Stack
+
+* Status: proposed
+* Deciders: Lauren Zugai, Jody Heavener, [TBD]
+* Date: 2020-09-08
+
+## Context and Problem Statement
+
+The front-end for Experimenter is currently composed of two applications. A more mature application for Normandy experiments, found at `app/experimenter/static/core`, is composed of a mix of Django templates, vanilla JS, jQuery, and React components. A newer React application for Nimbus experiments (previously named Rapid experiments) can be found at `app/experimenter/static/rapid`.
+
+With the decision made to build the Nimbus console as part of the Experimenter codebase ([see ADR #0001](https://github.com/mozilla/experimenter/blob/main/app/experimenter/docs/adrs/0001-build-nimbus-console-in-experimenter.md)), it was determined we likely want a "new front-end." This ADR, while slightly atypical as it does not serve to document a single architectural change, outlines a proposed directory tree change regarding the front-end, a proposed new front-end stack, how it compares to the current `rapid` stack, and how we will benefit from these changes.
+
+## Decision Drivers
+
+Drivers for directory tree changes:
+* A more obvious separation of concerns for:
+  * the front-end and back-end
+  * legacy front-end code and Nimbus front-end code
+
+Drivers for the proposed new stack:
+* Better development tooling
+* A more cohesive fit into the FxA team's stack and processes
+* Development speed
+
+Drivers for starting a new application:
+* Time needed and ease of refactoring the current `rapid` React application vs time needed to start with a new application
+
+## Proposed Changes
+
+### App Directory Tree
+
+Currently, a `static/` directory contains the front-end applications with some shared assets, and a `templates/` directory lives adjacent to it.  To provide a more obvious separation of concerns in the directory tree:
+
+* Rename `static/` to `legacy-ui/` since `static/rapid/` will become obsolete and `static/core/` will be the legacy Normandy platform
+* Move `templates/` into `legacy-ui/`
+* Use `create-react-app` to create a new React application in `app/experimenter/nimbus-ui/`
+
+
+### Front-End Stack
+
+* Continue using React, but:
+  * Use the `create-react-app` toolchain
+  * Use Typescript over Proptypes for more strict typing
+  * Use GraphQL for API calls and `apollo-client-cache` for global state management
+  * Use `reach-router` instead of `react-router`
+  * Use directories for components instead of filenames
+    * The Storybook and test file will live adjacent to the component `index.tsx` file. This allows all associated component files to live in the same directory and makes imports shorter.
+* Continue using Bootstrap (`react-bootstrap`), but:
```

Comment:
What's the strategy for including custom css when necessary?

Reply:
Oh yes +1 to this. With Tailwind we determined that there is almost no reason to use truly custom CSS because a utility class existed for every need (unless the custom selector is being used to stack utility classes). @LZoog because we'll undoubtedly need utility classes, I would say a con might be not being able to hook into PurgeCSS.

Reply:
I mostly included this piece because 1) under the `core` Normandy UI, we have files like `adminextra.css` where overrides are added and 2) `react-bootstrap` offers some customization options in SCSS like so: https://react-bootstrap.github.io/getting-started/introduction/#customize-bootstrap

Any custom SCSS should be kept to a minimum for maintenance purposes. 

>  I would say a con might be not being able to hook into PurgeCSS.

Ah yeah, I'll add that as an extension of this con I have for Bootstrap: "* Is a little bulky - we won’t use everything it offers, but this is a much less important consideration for an internal tool like Nimbus"

Reply:
Oh actually, it might be a bit of work, but [here](https://lukelowrey.com/automatically-removeunused-css-from-bootstrap-or-other-frameworks/) are instructions on how to set up PurgeCSS with Bootstrap. Looks like we'd need access to Webpack though.

Reply:
Eh yeah, I'm not sure it'd be worth it since it's an internal tool - we'd save like what, 50-100KB max? For a user-facing app I'd say let's do it for sure but I'm less inclined here to spend the effort setting it up. Worth filing as a "nice to have" though.

Reply:
Rescripts would be helpful. But I also agree that the time to implement may not be worth it initially. Seems like a good candidate for some downtime or as a refinement.